frontend/src/concepts/pipelines/content/createRun/contentSections/ParamsSection.tsx

```diff
@@ -4,58 +4,41 @@ import {
   CreateRunPageSections,
   runPageSectionTitles,
 } from '~/concepts/pipelines/content/createRun/const';
-import { RunParam } from '~/concepts/pipelines/content/createRun/types';
+import { RuntimeConfigParameters } from '~/concepts/pipelines/kfTypes';
 
 type ParamsSectionProps = {
-  value?: RunParam[];
-  onChange: (params: RunParam[]) => void;
+  runParams: RuntimeConfigParameters | undefined;
+  versionId: string | undefined;
+  onChange: (params: RuntimeConfigParameters) => void;
 };
 
-const ParamsSection: React.FC<ParamsSectionProps> = ({ value, onChange }) => {
+const ParamsSection: React.FC<ParamsSectionProps> = ({ runParams = {}, versionId, onChange }) => {
   const renderContent = (): React.ReactNode => {
-    if (!value) {
+    if (!versionId) {
       return (
         <Alert
           variant="info"
           isInline
           isPlain
-          title="You must select a pipeline before you can set parameters."
+          title="You must select a pipeline and version before you can set parameters."
         />
       );
     }
 
-    if (value.length === 0) {
-      return (
-        <Alert
-          component="h2"
-          variant="info"
-          isInline
-          isPlain
-          title="The selected pipeline has no input parameters."
-        />
-      );
+    if (!Object.keys(runParams).length) {
+      return <Alert variant="info" isInline isPlain title="This pipeline has no parameters." />;
     }
 
-    const handleChange = (index: number, newValue: RunParam) => {
-      const newParams = [...value];
-      newParams[index] = newValue;
-      onChange(newParams);
-    };
-    return (
-      <>
-        {value.map(({ label, value }, i) => (
-          <FormGroup key={label} label={label} fieldId={`${label}-param-field`}>
-            <TextInput
-              type="text"
-              id={`${label}-param-field`}
-              name={`${label}-param-field`}
-              value={value ?? ''}
-              onChange={(e, newValue) => handleChange(i, { label, value: newValue })}
-            />
-          </FormGroup>
-        ))}
-      </>
-    );
+    return Object.entries(runParams).map(([label, value]) => (
+      <FormGroup key={label} label={label} fieldId={`${label}-param-field`}>
+        <TextInput
+          id={`${label}-param-field`}
+          name={`${label}-param-field`}
+          value={value?.toString()}
```

Comment:
```suggestion
          value={value?.toString() ?? ''}
```
I think you need to do this to make sure React picks up the controlled nature of the input -- at least that was once upon a time in an earlier version of React.

Reply:
I haven't experienced any issues without this condition, but since it was there prior to my change I'll just add it back, especially since it doesn't hurt to add it.